vllm/engine/llm_engine.py

```diff
@@ -323,9 +334,12 @@ def _init_tokenizer(self, **tokenizer_init_kwargs):
             trust_remote_code=self.model_config.trust_remote_code,
             revision=self.model_config.tokenizer_revision)
         init_kwargs.update(tokenizer_init_kwargs)
+
         self.tokenizer = get_tokenizer_group(
             self.parallel_config.tokenizer_pool_config, **init_kwargs)
 
+        return self.tokenizer
```

Comment:
Since `self.tokenizer` has a type of `Optional[BaseTokenizerGroup]` (see above), a type error occurs when trying to pass it to `Detokenizer`. Therefore I am returning `BaseTokenizerGroup` from `_init_tokenizer` to ensure that `tokenizer` is not `None`.